src/test/java/de/retest/recheck/review/ignore/io/InheritanceLoaderTest.java

```diff
@@ -0,0 +1,94 @@
+package de.retest.recheck.review.ignore.io;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class InheritanceLoaderTest {
+
+	Loader<A> cut;
+
+	final A a = new A();
+	final B b = new B();
+	final C c = new C();
+	final Z z = new Z();
+
+	@BeforeEach
+	void setUp() {
+		final Loader<A> aLoader = mock( Loader.class );
+		when( aLoader.load( "a" ) ).thenReturn( Optional.of( a ) );
+		when( aLoader.save( a ) ).thenReturn( "a" );
+
+		final Loader<B> bLoader = mock( Loader.class );
+		when( bLoader.load( "b" ) ).thenReturn( Optional.of( b ) );
+		when( bLoader.save( b ) ).thenReturn( "b" );
+
+		final Loader<C> cLoader = mock( Loader.class );
+		when( cLoader.load( "c" ) ).thenReturn( Optional.of( c ) );
+		when( cLoader.save( c ) ).thenReturn( "c" );
+
+		cut = new InheritanceLoader<>( Arrays.asList( //
+				Pair.of( B.class, bLoader ), //	
+				Pair.of( C.class, cLoader ), // 	
+				Pair.of( A.class, aLoader ) //
+		) );
+	}
+
+	@Test
+	void load_should_parse_correct_strings() throws Exception {
+		assertThat( cut.load( "a" ) ).hasValue( a );
+		assertThat( cut.load( "b" ) ).hasValue( b );
+		assertThat( cut.load( "c" ) ).hasValue( c );
+		assertThat( cut.load( "d" ) ).isNotPresent();
+	}
+
+	@Test
+	void load_stream_should_parse_correct_strings() throws Exception {
+		assertThat( cut.load( Stream.of( "a", "b", "c", "d" ) ) ).contains( a, b, c );
+		assertThat( cut.load( Stream.of( "a", "a", "b", "c" ) ) ).contains( a, a, b, c );
+	}
+
+	@Test
+	void save_should_produce_correct_strings() throws Exception {
+		assertThat( cut.save( a ) ).isEqualTo( "a" );
+		assertThat( cut.save( b ) ).isEqualTo( "b" );
+		assertThat( cut.save( c ) ).isEqualTo( "c" );
+	}
+
+	@Test
+	void save_stream_should_produce_correct_strings() throws Exception {
+		assertThat( cut.save( Stream.of( a, b, c ) ) ).contains( "a", "b", "c" );
+		assertThat( cut.save( Stream.of( a, a, b, c ) ) ).contains( "a", "a", "b", "c" );
+	}
+
+	@Test
+	void save_with_unsupported_loader_should_produce_exception() throws Exception {
+		assertThatThrownBy( () -> cut.save( z ) ) //
+				.isInstanceOf( UnsupportedOperationException.class ) //
+				.hasMessage( String.format( "Did not find a loader for %s.", z ) );
+	}
+
+	@Test
+	void save_stream_with_unsupported_loader_should_produce_exception() throws Exception {
+		assertThatThrownBy( () -> cut.save( Stream.of( a, z, b ) ).count() ) // We need a finalizer here
```

Comment:
Two things:

* Doesn't throw an exception: https://travis-ci.com/retest/recheck/builds/135640725#L3163
* Maybe convert your comment into an actual `TODO` and elaborate what you have in mind.

Reply:
> Maybe convert your comment into an actual TODO

This was intentionally and is no TODO. It should have noted that the `save` only does a stream intermediate step; it does not evaluate the stream. Thus a finalizing method is needed, so that the stream is evaluated. I updated the comment to be hopefully more descriptive.

> Doesn't throw an exception

My mistake probably was that `count` might not evaluate the stream properly, if it can be retrieved by an other method (i.e. backing `list.size()`) (see [here](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/stream/Stream.html#count())). This seems to be an difference between my OracleJDK and the OpenJDK used in tests. I changed the finalizer to a `toArray` which must evaluate the Stream properly.